static/vendor/skulpt-stdlib-extensions.js

```diff
@@ -33,7 +33,15 @@ var $builtinmodule = function (name) {
             // If mapped key is a variable (not char), we retrieve variable value and use that
             // otherwise if char, use that.
             const charOrVar = value[0].v;
-            let mapLetterKey = Object.hasOwn(Sk.globals, charOrVar) ? Sk.globals[charOrVar].v : charOrVar;
+
+            let mapLetterKey = charOrVar;
```

Comment:
I think this solution is great and works, my only question would be if someone assigns a multileter string, and not a single letter, and then uses it in the key press. Like this:

```
x is asdf
if x is pressed
    print 'great'
else
    print 'not great'
```

Then it will always be false!

Reply:
You are right. We allow the left-hand-side of the if-pressed command to be a variable. And this variable could be assigned anything:  `x is 21 + 1` or `x is hello world`. At the same time we do not allow for multiple keys to be pressed at the same time.

What do you think is the best thing to do here? We could either silently take the first letter of the string which might be confusing. Alternatively, we could issue a warning at runtime (not sure if we have the mechanism for that?) that lets the user know that it is a multi-letter string so it will not work.

Since this issue was not introduced by the PR, I created a separate issue to discuss and track the matter: #5727

Reply:
I think the best way would be show an alert or an error warning the user that probably they don't want to be using a multicharacter string in a keypress! It can probably be done by checking the value of the variable and then calling the `modal.alert` function I think.